src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs

```diff
@@ -52,11 +50,13 @@ internal SynthesizedSimpleProgramEntryPointSymbol(SimpleProgramNamedTypeSymbol c
                     break;
             }
 
+            bool isNullableEnabled = compilation.IsNullableAnalysisEnabledIn(declaration.SyntaxReference.GetSyntax());
```

Comment:
>declaration.SyntaxReference.GetSyntax() [](start = 77, length = 39)

Is this entire compilation unit? I think this should be limited to global statements